kubernetes/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala

```diff
@@ -97,94 +97,32 @@ private[spark] class KubernetesClusterSchedulerBackend(
 
   def createExecutorPod(executorNum: Int): String = {
     // create a single k8s executor pod.
-    var annotationMap = Map(
-      "pod.beta.kubernetes.io/init-containers" -> raw"""[
-                {
-                    "name": "client-fetch",
-                    "image": "busybox",
-                    "command": ["wget", "-O", "/work-dir/client.jar", "$clientJarUri"],
-                    "volumeMounts": [
-                        {
-                            "name": "workdir",
-                            "mountPath": "/work-dir"
-                        }
-                    ]
-                },
-                {
-                    "name": "distro-fetch",
-                    "image": "busybox",
-                    "command": ["wget", "-O", "/work-dir/spark.tgz", "$sparkDistUri"],
-                    "volumeMounts": [
-                        {
-                            "name": "workdir",
-                            "mountPath": "/work-dir"
-                        }
-                    ]
-                },
-                {
-                    "name": "setup",
-                    "image": "$sparkDriverImage",
-                    "command": ["./install.sh"],
-                    "volumeMounts": [
-                        {
-                            "name": "workdir",
-                            "mountPath": "/work-dir"
-                        },
-                        {
-                            "name": "opt",
-                            "mountPath": "/opt"
-                        }
-                    ]
-                }
-            ]""")
-
-
     val labelMap = Map("type" -> "spark-executor")
     val podName = s"$sparkExecutorName-$executorNum"
     var pod = new PodBuilder()
       .withNewMetadata()
       .withLabels(labelMap.asJava)
       .withName(podName)
-      .withAnnotations(annotationMap.asJava)
       .endMetadata()
       .withNewSpec()
       .withRestartPolicy("OnFailure")
+
       .addNewContainer().withName("spark-executor").withImage(sparkDriverImage)
-      .withImagePullPolicy("Always")
-      .withCommand("/opt/spark/bin/spark-class")
+      .withImagePullPolicy("IfNotPresent")
+      .withCommand("/opt/executor.sh")
```

Comment:
Another thing that was problematic with the init-containers popped up when running behind a proxy server. The wget command used to fetch the jar from the network url into to the specified path would fail and other init-containers won't get created and consequently the pod as well.
With a prebuilt spark image (with the necessary entry point shims), you could easily build your own docker images with --build-args=<http/https_proxy>